sympy/tensor/array/tests/test_array_comprehension.py

```diff
@@ -58,9 +58,11 @@ def test_arraycomprehensionmap():
     assert a.tolist() == [2, 3, 4, 5, 6]
     assert len(a) == 5
     assert isinstance(a.doit(), ImmutableDenseNDimArray)
+    expr = ArrayComprehensionMap(lambda i: i+1, (i, 1, k))
+    assert expr.doit() == expr
+    assert expr.subs(k, 4) == ArrayComprehensionMap(lambda i: i+1, (i, 1, 4))
```

Comment:
`.subs( ).doit()` as well. It looks like the `.func` trick works.

Reply:
Yes! Thanks for this instruction!